### PR TITLE
DHFPROD-4158: Can now deploy to DHS via multiple roles

### PR DESCRIPTION
**Pull request by rjrudin**
Most of the files in here are for a new example project - dhs-example. 

This is now simpler as we're no longer in the business of eagerly setting DHS-specific properties, like basic auth and SSL. It'll still be up to the user to set those in their gradle-dhs.properties file.

### CODE REVIEW

File: examples/dhs-example/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+As of 5.2.0, a user with the data-hub-security-admin role is permitted to deploy roles that grant privileges that are
+inherited by the user performing the deployment. As an example of this, the 
+src/main/ml-config/security/roles/custom-role1.json file defines a new role with the "manage" privilege, which is 
+inherited by the data-hub-security-admin role. 
+
+Permitted resources can be deployed via the following task (assuming that gradle-dhs.properties defines the host and
+credentials for the DHS instance):
+
+    ./gradlew -PenvironmentName=dhs dhsDeployAsSecurityAdmin
```
Comment (bsrikan) [inline examples/dhs-example/README.md]: Should this be changed to hubDeployAsSecurityAdmin?

File: examples/dhs-example/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+## Scheduled tasks
+
+Because a DHS instance uses multiple groups for application servers, and because a scheduled task is associated with a
+group, scheduled task files are stored in the directory "./tasks/(name of group)". See 
+src/main/ml-config/tasks for an example in this project. Also see
+[the ml-gradle docs](https://github.com/marklogic-community/ml-gradle/wiki/Resource-reference#scheduled-tasks) for more information.
+
+## Temporal axes and collections
+
```
Comment (bsrikan) [inline examples/dhs-example/README.md]: Is it worth mentioning here that lsqt is not supported?

File: ml-data-hub-plugin/src/main/groovy/com/marklogic/gradle/DataHubPlugin.groovy [hunk 3]
```diff
@@ -190,10 +184,18 @@ class DataHubPlugin implements Plugin<Project> {
...
+
+        project.task("hubDeployAsDeveloper", group: deployGroup, type: DeployAsDeveloperTask,
+            description: "Deploy project configuration as a user with the data-hub-developer role"
+        ).finalizedBy(["hubGenerateExplorerOptions"]).mustRunAfter("hubDeployAsSecurityAdmin")
+
+        project.task("hubDeploy", group: deployGroup, dependsOn: ["hubDeployAsDeveloper", "hubDeployAsSecurityAdmin"],
+            description: "Deploy project configuration as a user with the data-hub-security-admin and data-hub-developer roles")
+
+        project.task("dhsDeploy", dependsOn: ["hubDeploy"],
+            description: "dhsDeploy has been replaced in 5.2.0 by hubDeploy and similar tasks. It is now simply an alias for hubDeploy.")
```
Comment (bsrikan) [inline ml-data-hub-plugin/src/main/groovy/com/marklogic/gradle/DataHubPlugin.groovy]: nice addition.

File: examples/dhs-example/README.md (hunk 1 above)
Comment (rjrudin) [reply examples/dhs-example/README.md]: Good catch, made the fix.

File: examples/dhs-example/README.md (hunk 2 above)
Comment (rjrudin) [reply examples/dhs-example/README.md]: That's a good idea,  I added mention of that.

**Pull request merged**